### ISSUE TITLE
Mispellings in main_file aren't caught

### ISSUE THREAD

**Issue by facundominguez** [tweag/rules_haskell#1637]
When writing a test where the main module is not called Main.hs, we need to specify the `main_file` attribute.

```python
haskell_test(
    name = "binary-with-main-file",
    srcs = [
        "src/MainIsHere.hs",
        "src/M.hs",
    ],
    main_file = "rc/MainIsHere.hs",
    deps = ["//tests/hackage:base"],
)
```

However, if `main_file` has a misspelling or is not listed in `srcs`, bazel will complain that some outputs were not generated.
```
INFO: Invocation ID: 38a4c8a1-1eee-401a-9c84-1a8c38429208
INFO: Analyzed 3 targets (1 packages loaded, 10 targets configured).
INFO: Found 1 target and 2 test targets...
ERROR: /home/facundo/tweag/rules_haskell/tests/binary-with-main/BUILD.bazel:16:13: output 'tests/binary-with-main/_obj/binary-with-main-file/MainIsHere.o' was not created
ERROR: /home/facundo/tweag/rules_haskell/tests/binary-with-main/BUILD.bazel:16:13: output 'tests/binary-with-main/_iface/binary-with-main-file/MainIsHere.hi' was not created
ERROR: /home/facundo/tweag/rules_haskell/tests/binary-with-main/BUILD.bazel:16:13: HaskellBuildBinary //tests/binary-with-main:binary-with-main-file failed: not all outputs were created or valid
ERROR: /home/facundo/tweag/rules_haskell/tests/binary-with-main/BUILD.bazel:16:13 HaskellLinkBinary tests/binary-with-main/binary-with-main-file failed: not all outputs were created or valid
INFO: Elapsed time: 1.168s, Critical Path: 1.01s
INFO: 4 processes: 3 internal, 1 linux-sandbox.
FAILED: Build did NOT complete successfully
//tests/binary-with-main:binary-with-main                             NO STATUS
//tests/binary-with-main:binary-with-main-file                  FAILED TO BUILD

FAILED: Build did NOT complete successfully
```

The expected behavior is that `rules_haskell` produces an error saying that `main_file` must appear listed in `srcs`.
